book/src/volume-expansion.md

```diff
@@ -0,0 +1,46 @@
+# Volume Expansion
+
+**Status:** Alpha
+
+Many storage providers allow a volume to be expanded after creation. A storage provider may choose to implement volume expansion either via a
+control-plane CSI RPC call or via node CSI RPC call or both as a two step process.
+
+## Implementing Volume expansion functionality
+
+To implement volume expansion the CSI driver MUST:
+
+1. Implement `VolumeExpansion` plugin capability.
+2. Implement `EXPAND_VOLUME` controller capability or implement `EXPAND_VOLUME` node capability or both.
+
+`ControllerExpandVolume` RPC call can be made when volume is `ONLINE` or `OFFLINE` depending of `VolumeExpansion` plugin
+capability.
+
+`NodeExpandVolume` RPC call on the other hand - *always* requires volume to be published or staged on a node (and hence `ONLINE`).
```

Comment:
Is it called twice? Once after NodeStage, and once after NodePublish?

Reply:
nope. just once - I will try and clarify.

Reply:
Thinking some more, I think I would not want this document to serve as a replacement of CSI specs. I think when `NodeExpandVolume` is called is obvious in CSI's `spec.md` and I would not want to repeat that information.